modules/pipeline/pipengine/actionexecutor/plugins/docker/docker.go

```diff
@@ -0,0 +1,425 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/pipeline/pipengine/actionexecutor/logic"
+	"github.com/erda-project/erda/modules/pipeline/pipengine/actionexecutor/types"
+	"github.com/erda-project/erda/modules/pipeline/pkg/task_uuid"
+	"github.com/erda-project/erda/modules/pipeline/spec"
+	"github.com/erda-project/erda/pkg/strutil"
+)
+
+var Kind = types.Kind(spec.PipelineTaskExecutorKindDocker)
+
+func init() {
+	types.MustRegister(Kind, func(name types.Name, options map[string]string) (types.ActionExecutor, error) {
+		return New(name)
+	})
+}
+
+// TODO: create docker client by executor-enviroment
+// the remote docker client should be created by host, tls-veiry like client.WithHost("tcp://xxx:2375")
+// the local docker  client could be created by docker-sock env directly like clinet.FromEnv
+func New(name types.Name) (*DockerJob, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create docker client")
+	}
+	return &DockerJob{
+		name:       name,
+		client:     cli,
+		errWrapper: logic.NewErrorWrapper(name.String()),
+	}, nil
+}
+
+type DockerJob struct {
+	client     *client.Client
+	name       types.Name
+	errWrapper *logic.ErrorWrapper
+}
+
+func (d *DockerJob) Kind() types.Kind {
+	return Kind
+}
+
+func (d *DockerJob) Name() types.Name {
+	return d.name
+}
+
+func (d *DockerJob) Status(ctx context.Context, task *spec.PipelineTask) (desc apistructs.PipelineStatusDesc, err error) {
+	if err := logic.ValidateAction(task); err != nil {
+		return desc, err
+	}
+	jobName := logic.MakeJobName(task)
+	container, err := d.client.ContainerInspect(ctx, jobName)
+	if err != nil {
+		return desc, err
+	}
+	switch container.State.Status {
+	case "running", "restarting", "created":
```

Comment:
does these status have const or typed status?

Reply:
not found in docker pkg
![image](https://user-images.githubusercontent.com/30427474/162684026-5c9ee742-ca57-4a34-99ec-e4a751fc1cd5.png)